plugins/experimental/cachekey/plugin.cc

```diff
@@ -91,8 +91,28 @@ TSRemapDoRemap(void *instance, TSHttpTxn txn, TSRemapRequestInfo *rri)
   Configs *config = (Configs *)instance;
 
   if (nullptr != config) {
+    TSMBuffer bufp;
+    TSMLoc urlLoc;
+    TSMLoc hdrLoc;
+
+    /* Get the URI and header to base the cachekey on, use the remap URI by default or if any failure.
+     * @TODO it might make sense to add more supported URI types */
+    bool useDefault = true;
+    if (PRISTINE == config->getUriType()) {
+      if (TS_SUCCESS == TSHttpTxnPristineUrlGet(txn, &bufp, &urlLoc)) {
```

Comment:
If for whatever reason TSHttpTxnPristineUrlGet() fails, I don't think we should fall back to the "REMAP" configuration case. That seems like it'd be potentially dangerous, and at least confusing .-).

Reply:
@zwoop good point, will change that, thanks!